Jetcaster/app/src/main/java/com/example/jetcaster/ui/player/PlayerViewModel.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetcaster.ui.player
+
+import android.net.Uri
+import android.os.Bundle
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.AbstractSavedStateViewModelFactory
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.savedstate.SavedStateRegistryOwner
+import com.example.jetcaster.Graph
+import com.example.jetcaster.data.EpisodeStore
+import com.example.jetcaster.data.PodcastStore
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import java.time.Duration
+
+data class PlayerUiState(
+    val title: String = "",
+    val duration: Duration? = null,
+    val podcastName: String = "",
+    val podcastImageUrl: String = ""
+)
+
+class PlayerViewModel(
+    episodeStore: EpisodeStore,
+    podcastStore: PodcastStore,
+    savedStateHandle: SavedStateHandle
+) : ViewModel() {
+
+    private val episodeUri: String = Uri.decode(savedStateHandle.get<String>("episodeUri")!!)
```

Comment:
this lack of error handling makes me nervous, at least add a comment?

Reply:
Added! Thanks